mitodl/open-discussions
# Course search UI

#### Pre-Flight checklist



#### What are the relevant tickets?
Closes #1719 

#### What's this PR do?
- Adds `SETTINGS.course_ui_enabled`
- Adds a `/course/search` URL to the App router if the above is `true`
- Adds a `CourseSearchPage` container and associated components & functions

#### How should this be manually tested?
- Set `FEATURE_COURSE_UI=True` in your `.env` file
- Set `EDX_*` and `OCW_*` .env variables
- If you haven't done so recently, run `recreate_index`
- In a django shell, run `from course_catalog.tasks import get_edx_data;get_edx_data()`
- You can also run `get_ocw_data()` and stop after a few are imported. 
- Go to `/courses/search` and try it out ("mechanical" is a good sample query), with and without facets checked.  
- As you change search parameters, the browser's URL should update.  Copying and pasting the URL into a new browser tab should load the same results.
-  Set `FEATURE_COURSE_UI=False` in your `.env` file and restart docker
- Go to `/courses/search`, you should get a blank page

#### Screenshots (if appropriate)

*Desktop, various widths:*

![screen shot 2019-02-12 at 2 34 55 pm](https://user-images.githubusercontent.com/187676/52663942-d3edce80-2ed5-11e9-99c5-8d1942ea381f.png)

![screen shot 2019-02-12 at 2 35 11 pm](https://user-images.githubusercontent.com/187676/52663943-d7815580-2ed5-11e9-9319-47b60573800a.png)

![screen shot 2019-02-12 at 2 35 26 pm](https://user-images.githubusercontent.com/187676/52663946-da7c4600-2ed5-11e9-8836-8f96d9324240.png)

![screen shot 2019-02-12 at 2 35 45 pm](https://user-images.githubusercontent.com/187676/52663951-dd773680-2ed5-11e9-88d1-6bae70517d86.png)


*Mobile view:*

![screen shot 2019-01-30 at 12 32 20 pm](https://user-images.githubusercontent.com/187676/52000925-426d6e00-248c-11e9-82a5-7d32f256132f.png)

![screen shot 2019-02-12 at 2 36 16 pm](https://user-images.githubusercontent.com/187676/52663983-ee27ac80-2ed5-11e9-858a-7d391ab13901.png)
